docs/how-to-guides/construct-test-data.md

```diff
@@ -0,0 +1,96 @@
+# How to construct test data based on documents
+This guide will instruct you on how to generate test data for RAG systems utilizing existing documents.
+Previously evaluating the performance of RAG systems required the creation of test data. This could be done manually, a process that required significant time and effort, or by purchasing pre-made test data, which could be costly.
+
+This test data generation process streamlines this by leveraging the capabilities of llm to automatically generate the test data. This not only reduces the effort required but also eliminates the need for additional expenditures.
+
+By following this guide, you will learn how to:
+1. Customize the test data generation process by tuning flow prompts.
+2. Generate high-quality test data quickly and easily by running a test data generation script.
+
+**Supported file types**
+- .md - Markdown
+- .docx - Microsoft Word
+- .pdf - Portable Document Format
+- .ipynb - Jupyter Notebook
+
+**Limitations**
+
+- While the test data generator works well with standard documents, it may face challenges with API introduction documents or reference documents.
+- Additionally, it may not function effectively for non-Latin characters, such as Chinese. These limitations may be due to the text loader capabilities such as pypdf.
+
+## Quick start
+- Install required packages
+Enter `test_data_gen_local` folder, run below command to install required packages.
+  ```bash
+  pip install -r requirements.txt
+  ```
+
+  For specific document file types, you will need to install extra packages:
+  > !Note: This package requirement may be outdated. In this process, we utilize llama index `SimpleDirectoryReador`. For the most recent information, please check [here](https://docs.llamaindex.ai/en/stable/examples/data_connectors/simple_directory_reader.html).
+  - .docx - `pip install docx2txt`
+  - .pdf - `pip install pypdf`
+  - .ipynb - `pip install nbconvert`
+
+- Run and tune test data construction flow
+  - Navigate to the [construct_test_data_flow folder](../../examples/test_data_gen/construct_test_data_flow/). Open the `flow.dag.yaml` to understand the structure of the data flow. Fill in necessary values like connections and model/deployment name.
+  - Ensure the `flow.tools.json` is generated under `.promptflow` folder within the flow folder.
```

Comment:
Why?

Reply:
When submit flow as component in cloud, the flow.tools.json is required

Reply:
It's expected that only UX has dependency on flow.tools.json. 

Please share detailed failure message and verify if this is a sdk limitation/bug.